### ISSUE TITLE
Success Criterion 3.2.3 - Consistent Navigation - Level AA

### ISSUE THREAD

**Issue by JJdeGroot** [w3c/matf#54]
WCAG2ICT guidance: https://www.w3.org/TR/wcag2ict-22/#consistent-navigation

Share your thoughts for applying to mobile apps as a comment below.

**Comment by JJdeGroot** [comment]
Discussed in yesterday's meeting. Will be on agenda again next week.

<details>
<summary>9 October 2024</summary>

[Source](https://www.w3.org/2024/10/09-matf-minutes.html#29d3)

~~~text
JJ: WCAG2ICT added a note that while not required SC should be considered best practice
… relates to the problematc definition to "set of software programs" (rather than "set of views")

JJ: In the EN 301 549 it is also considered to change to sets within software

Jamie: If we have a note, the way it is written does not clearly stating the purpose
… seems confusing right now - we should not use the same wording

JJ: if we can view or screen it will be clearer

JJ: We might need an exception because there may be valid reasons to deviate in terms of order
… need ti figure out if exception is needed - may not be needed since note is not normative

JJ: we may also discuss consistent help - applies more to web sites

JJ: will send out minutes after the meeting
~~~
</details>

**Comment by JJdeGroot** [comment]
- [Reconsidering WCAG-based software requirements omitted from EN 301 549 v.3.2.1](https://labs.etsi.org/rep/HF/en301549/-/issues/278)
- [EN 301 549 discussion for 3.2.3 Consistent Navigation](https://labs.etsi.org/rep/HF/en301549/-/issues/281)

Most recent proposal:
> Navigational mechanisms that are repeated within a software program occur in the same relative order each time they are repeated, unless a change is initiated by the user or where not doing so is essential to the function of the software.

This replaces "set of web pages" with "a software program"
I proposed to add: _"except where not doing so is essential to the function of the software."_ to add the same exception that was accepted for [3.2.4 Consistent Identification](https://github.com/w3c/matf/issues/55)

I think we should phrase our guidance similarly, ideally with "views" or "screens". Or we could phrase it as "a mobile application" to apply it to all content contained in the application.

**Comment by JJdeGroot** [comment]
Discussed in today’s meeting.

<details>
<summary>16 October 2024</summary>

[Source](https://www.w3.org/2024/10/16-matf-minutes.html#caab)

~~~text
JJ - WCAG2ICT replaces "set of web pages" with "a software program"

JJ - correction - EN 301 549 replaces "set of web pages" with "a software program"

JJ - This mostly related to how we replace "set of web pages".

<JJ> Mick: in our group we would look at a relatively small set of components for consistent navigations, such as search buttons?

JJ - I would consider a back button, close button, buttons in a list, a change of tab in a tab bar.

Aash - should we introduce certain things - the bottom sheets without the close button. Very hard to dismiss those overlays.

JJ - I think it would be interesting to add such things in a Note. That might be more related to Keyboard Operation rather than this success criterion

JJ - I wonder would a back button or close button placed differently would fail this criteria in app or is it too strict. Is it more about large lists of links rather than a single link/button.

JJ - Does 'navigational mechanisms' related to one button or link

Jamie - this criteria isn't about where the back button is on every app

ACTION: Define "navigational mechanisms"

Jamie - what are we considering navigational mechanisms? It would be helpful outlining examples.

JJ - this is specific to a single app.
~~~
</details>

**Comment by JJdeGroot** [comment]
**[Text in EN 301 549 v4.1.1](https://www.etsi.org/deliver/etsi_en/301500_301599/301549/04.01.01_60/en_301549v040101p.pdf#page=99)**

> 11.3.2.3 Consistent navigation (Void)
> 
> **NOTE:** WCAG 2.2 Success Criterion 3.2.3 "Consistent navigation" applies to "Sets" of web pages. While
> consistency within non-web software is desirable, "sets of non-web software" in the same sense as "sets
> of web pages", are extremely rare and an equivalent is not included in this clause on non-web software
> requirements.

**[Text in WCAG2ICT Editor's Draft (18 September 2026)](https://w3c.github.io/wcag2ict/#consistent-navigation)**

> This applies directly as written, and as described in [Intent from Understanding Success Criterion 3.2.3](https://www.w3.org/WAI/WCAG22/Understanding/consistent-navigation#intent), replacing "on multiple web pages within a set of web pages" with "in multiple non-web software programs within a set of non-web software programs”.
> 
> With these substitutions, it would read:
> 
> 3.2.3 Consistent Navigation: Navigational mechanisms that are repeated [in multiple [non-web software programs](https://w3c.github.io/wcag2ict/#software) within a [set of non-web software programs](https://w3c.github.io/wcag2ict/#set-of-software-programs)] occur in the [same relative order](https://www.w3.org/TR/WCAG22/#dfn-same-relative-order) each time they are repeated, unless a change is initiated by the user.
>
> **Note 1 (Added)**
> 
> See [set of software programs](https://w3c.github.io/wcag2ict/#set-of-software-programs) in the Key Terms section to determine when a group of software programs is considered a set for this success criterion. Those implementing this document (WCAG2ICT) will need to consider if this success criterion is appropriate to apply to non-web software. See the [Interpretation of Web Terminology in a Non-web Context](https://w3c.github.io/wcag2ict/#interpretation-of-web-terminology-in-a-non-web-context).
> 
> **Note 2 (Added)**
> 
> Although not required by this success criterion, ensuring that navigation elements have consistent order when repeated within non-web software programs directly addresses user needs identified in the Intent section for this success criterion, and is generally considered best practice.
> 
> **Note 3 (Added)**
> 
> Sets of software that meet this definition appear to be extremely rare.
> 
> **Note 4 (Added)**
> 
> For general guidance, see also the [comments on closed functionality](https://w3c.github.io/wcag2ict/#comments-on-closed-functionality). For commentary on the individual success criterion, see [Success criteria problematic for closed functionality, Consistent Navigation](https://w3c.github.io/wcag2ict/#problematic-for-closed-consistent-navigation).
